packages/fast-components/src/checkbox/checkbox.ts

```diff
@@ -0,0 +1,167 @@
+import { attr, FastElement, observable, Observable } from "@microsoft/fast-element";
+import { FormAssociated } from "../form-associated";
+import { keyCodeSpace } from "@microsoft/fast-web-utilities";
+import { bool } from "../utilities";
+
+/* tslint:disable:member-ordering */
+export class Checkbox extends FormAssociated<HTMLInputElement> {
+    @attr({ attribute: "readonly" })
+    public readOnly: boolean; // Map to proxy element
+    private readOnlyChanged(): void {
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.readOnly = this.readOnly;
+        }
+
+        this.readOnly
+            ? this.classList.add("readonly")
+            : this.classList.remove("readonly");
+        this.setAttribute("aria-readonly", bool(this.readOnly).toString());

```

Comment:
As a follow-up we can use the new host directives feature to bind these attributes on the host element itself, which should remove all the manual aria attr code. Not required to change it now, but a nice incremental improvement.

Reply:
Implemented